dartagnan/src/main/java/com/dat3m/dartagnan/encoding/ProgramEncoder.java

```diff
@@ -277,65 +284,70 @@ private BooleanFormula encodeForwardProgress(Program program, ProgressModel prog
         };
     }
 
-    // =====================================================================================
+    // ====================================== Memory layout ======================================
 
-    // Encodes the address values of memory objects.
+    // Encodes the address values and sizes of memory objects.
     public BooleanFormula encodeMemory() {
         logger.info("Encoding memory");
-        final Memory memory = context.getTask().getProgram().getMemory();
-        final FormulaManager fmgr = context.getFormulaManager();
-        // TODO: Once we want to encode dynamic memory layouts (e.g., due to dynamically-sized mallocs)
-        //  we need to compute a Map<MemoryObject, Formula> where each formula describes a
-        //  unique, properly aligned, and non-overlapping address of the memory object.
-        final Map<MemoryObject, BigInteger> memObj2Addr = computeStaticMemoryLayout(memory);
-
-        final var enc = new ArrayList<BooleanFormula>();
-        for (final MemoryObject memObj : memory.getObjects()) {
-            // For all objects, their 'final' value fetched here represents their constant value.
-            final Formula addressVariable = context.encodeFinalExpression(memObj);
-            final BigInteger addressInteger = memObj2Addr.get(memObj);
-
-            if (addressVariable instanceof BitvectorFormula bitvectorVariable) {
-                final BitvectorFormulaManager bvmgr = fmgr.getBitvectorFormulaManager();
-                final int length = bvmgr.getLength(bitvectorVariable);
-                enc.add(bvmgr.equal(bitvectorVariable, bvmgr.makeBitvector(length, addressInteger)));
+        return encodeMemoryLayout(context.getTask().getProgram().getMemory());
```

Comment:
What is the point of creating a new method `encodeMemoryLayout` when `encodeMemory` is a 2-liner that simply logs one things and calls it.

Reply:
Because there were temporarily two different ways to encode memory, depending on whether it is static or not. The question is whether we need to bring back the static version due to some regressions and if we do that, whether we want to split the functions agains.
For now, I decided to keep only one variant but still have the split. I also think it is not too bad to have the public interface `encodeMemory` be separated from the internal implementation `encodeMemoryLayout` but I guess that is a matter of personal taste.